.github/workflows/linux.yml

```diff
@@ -27,7 +27,7 @@ jobs:
           environment-name: build_env
           cache-environment: true
       - name: Install external dependencies
-        run: micromamba install -n build_env -y -f ./environment-dev-external-deps.yml
+        run: micromamba install -n build_env -y -f ./environment-dev.yml
```

Comment:
I thought we wanted to separate tests with packages from conda and fetched from source? Do we eventually go with only using `DFETCH_DEPENDENCIES_WITH_CMAKE`?
In that case, this job is useless and needs to be removed (installating packages is already done just before).